scanpipe/pipes/d2d_config.py

```diff
@@ -0,0 +1,167 @@
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# Data Generated with ScanCode.io is provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+# ScanCode.io should be considered or used as legal advice. Consult an Attorney
+# for any legal advice.
+#
+# ScanCode.io is a free software code scanning tool from nexB Inc. and others.
+# Visit https://github.com/aboutcode-org/scancode.io for support and download.
+
+
+from dataclasses import dataclass
+from dataclasses import field
+
+
+@dataclass
+class EcosystemConfig:
+    """
+    Base class for ecosystem-specific configurations to be defined
+    for each ecosystem.
+    """
+
+    # This should be defined for each ecosystem which
+    # are options in the pipelines
+    ecosystem_option: str = "Default"
+
+    # These are extensions for packages of this ecosystem which
+    # need to be matched from purldb
+    purldb_package_extensions: list = field(default_factory=list)
+
+    # These are extensions for resources of this ecosystem which
+    # need to be matched from purldb
+    purldb_resource_extensions: list = field(default_factory=list)
+
+    # Extensions for document files which do not require review
+    doc_extensions: list = field(default_factory=list)
+
+    # Paths in the deployed binaries/archives (on the to/ side) which
+    # do not need review even if they are not matched to the source side
+    deployed_resource_path_exclusions: list = field(default_factory=list)
+
+    # Paths in the development/source archive (on the from/ side) which
+    # should not be considered even if unmapped to the deployed side when
+    # assessing what to review on the deployed side
+    devel_resource_path_exclusions: list = field(default_factory=list)
+
+    # Symbols which are found in ecosystem-specific standard libraries
+    # which are not so useful in mapping
+    standard_symbols_to_exclude: list = field(default_factory=list)
+
+
+# Dictionary of ecosystem configurations
+ECOSYSTEM_CONFIGS = {
+    "Default": EcosystemConfig(
+        purldb_package_extensions=[".zip", ".tar.gz", ".tar.xz"],
+        devel_resource_path_exclusions=["*/tests/*"],
+        doc_extensions=[
+            ".pdf",
+            ".doc",
+            ".docx",
+            ".ppt",
+            ".pptx",
+            ".tex",
+            ".odt",
+            ".odp",
+        ],
+    ),
+    "Java": EcosystemConfig(
+        ecosystem_option="Java",
+        purldb_package_extensions=[".jar", ".war"],
+        purldb_resource_extensions=[".class"],
```

Comment:
what about .aj, java, .kt, and so on? I guess they will be handled in:
- https://github.com/aboutcode-org/scancode.io/issues/1404

Reply:
Yeah, these will be handled in #1404, added when the support is added and tested.